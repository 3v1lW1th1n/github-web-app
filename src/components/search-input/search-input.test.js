import 'jest-styled-components'

import React from 'react'
import { shallow, mount } from 'enzyme'
import SearchInput from './'

describe('<SearchInput />', () => {
  it('should match default snapshot', () => {
    const component = shallow(<SearchInput />)
    expect(component).toMatchSnapshot()
  })

  it('should send a value to search', () => {
    const mockRequestFn = jest.fn()
    const searchquery = 'username'
    const component = mount(<SearchInput onRequest={mockRequestFn} />)
    component.find('input.search-input__field').instance().value = searchquery
    component.find('button.search-input__button').prop('onClick')()
    expect(mockRequestFn).toHaveBeenCalled()
    expect(mockRequestFn).toBeCalledWith(searchquery)
  })

  it('should have the right displayName', () => {
    expect(SearchInput.displayName).toEqual('SearchInput')
  })
})
