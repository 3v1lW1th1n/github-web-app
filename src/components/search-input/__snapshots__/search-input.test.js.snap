// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> should match default snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "_foldedDefaultProps": Object {
        "placeholder": "O que deseja pesquisar?",
        "value": "",
      },
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "StyledSearchInput-sc-5q7857-0",
        "isStatic": true,
        "rules": Array [
          "display:flex;.search-input__field{box-sizing:border-box;width:250px;height:48px;border:1px solid rgba(0,0,0,.3);border-radius:5px;padding:8px;&:focus{border-color:rgba(0,0,0,.75);}&::placeholder{text-align:center;}}.search-input__button{margin-left:8px;height:48px;width:48px;display:inline-flex;align-items:center;justify-content:center;background-color:#767676;color:#fff;border:1px solid #767676;border-radius:5px;}",
        ],
      },
      "displayName": "SearchInput",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "StyledSearchInput-sc-5q7857-0",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  placeholder="O que deseja pesquisar?"
  value=""
/>
`;
